### ISSUE TITLE
DATE-Problem: 0000-01-01 becomes 0001-01-01 (syntactically)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#15]
```
What steps will reproduce the problem?

1. create table with two rows
CREATE TABLE p(d DATE);
INSERT INTO p VALUES('0000-01-01');
INSERT INTO p VALUES('0001-01-01');

2.
SELECT * FROM p;

yields two identical rows

-----------|-
D          |
-----------|- 
0001-01-01 |
0001-01-01 |
-----------|-

3.
SELECT * FROM p WHERE d='0000-01-01'

at least returns only one row

-----------|-
D          |
-----------|- 
0001-01-01 |
-----------|-

... nevertheless with wrong representation. 

What is the expected output? What do you see instead?
0000-01-01 

What version of the product are you using? On what operating system, file
system, and virtual machine?

H2 Version: H2 1.0.67 (2008-02-22)
OS: WIN XP Pro Version 2002
Java: 1.6.0_05

Do you know a workaround?
No

How important/urgent is the problem for you?
minor

In your view, is this a defect or a feature request?
defect
```

Original issue reported on code.google.com by `frank.ka...@gmail.com` on 12 Mar 2008 at 8:06

**Comment by GoogleCodeExporter** [comment]
```
This looks like a display problem (The year is 1 BC).

MySQL, Derby: An exception is thrown trying to insert '0000-01-01'.

HSQLDB: The same behavior as H2

PostgreSQL: 0001-01-01 BC; 0001-01-01

```

Original comment by `thomas.t...@gmail.com` on 12 Mar 2008 at 10:42

**Comment by GoogleCodeExporter** [comment]
Original comment by `thomas.t...@gmail.com` on 12 Mar 2008 at 10:42
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
For years below 1, the YEAR method doesn't return the correct value, and the
conversion from date and timestamp to varchar is incorrect. This will be fixed 
in the
next release.

Actually this is a problem of java.sql.Date and java.sql.Timestamp.toString(): 
those
methods just ignore the 'BC'. One solution would be to disallow using years 
lower
than 1 like MySQL and Derby, but I don't like this. I will add a workaround for 
the
VARCHAR conversion of dates and timestamps below 1 (BC).

The behavior will be:
CREATE TABLE p(d DATE);
INSERT INTO p VALUES('-0001-01-01'), ('0000-01-01'), ('0001-01-01');
select d, year(d) y, extract(year from d) y2, cast(d as timestamp) ts from p;
D           Y   Y2  TS  
-1-01-01    -1  -1  -1-01-01 00:00:00.0
0-01-01     0    0  0-01-01 00:00:00.0
0001-01-01  1    1  0001-01-01 00:00:00.0

```

Original comment by `thomas.t...@gmail.com` on 21 Mar 2008 at 11:12

**Comment by GoogleCodeExporter** [comment]
```
Fixed in version 1.0.69 (2008-03-29)
```

Original comment by `thomas.t...@gmail.com` on 29 Mar 2008 at 12:30
- Changed state: **Fixed**
